src/Api/AbstractApi.php

```diff
@@ -271,10 +271,6 @@ private static function getContent(Response $response)
     {
         $content = ResponseMediator::getContent($response);
 
-        if (null === $content) {
-            throw new RuntimeException('No content was provided.');
-        }
-
-        return $content;
+        return (object)$content;
```

Comment:
Fixes issue #254 by returning the content casted to an object (so it's an object even when content is null) and not throwing an exception.
But i'm not sure that this is a great fix nor the best thing to do.

Reply:
I think the proper fix for this is to not call this entire method when there is no content. I intentionally added an exception to this method. We need to locate all places in the codebase where a 204 is expected, and change them to return void and not call this code.

Please revert your changes to this class in this PR.